src/utils/optionsparser.cc

```diff
@@ -405,17 +406,34 @@ void StringOption::SetVal(OptionsDict* dict, const ValueType& val) const {
 // IntOption
 /////////////////////////////////////////////////////////////////
 
+namespace {
+int ValidateIntString(const std::string& flag, const std::string& val) {  
+  int result;
+  const auto end = val.data() + val.size();
+  auto [ptr, err] = std::from_chars(val.data(), end, result);  
+  if (err == std::errc::invalid_argument) {
+    throw Exception("Flag '--" + flag + "' has an invalid format.");
```

Comment:
Oops I didn't notice it did call the member functions. In that case it would be find to keep it as they were (but make `const`).

Feel free to leave it as it is or undo that change (but then add `const`), as you prefer.